### ISSUE TITLE
[SCM-764] username and credentials shown as INFO on commadline

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#971]
**[Thomas Wabner](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=waffel)** opened **[SCM-764](https://issues.apache.org/jira/browse/SCM-764?redirect=false)** and commented

Using git repository with gitblit on HTTPS.

Every git command which involve the remote repository (like fetch, pull, push and so on) showing the username and credentials on the commandline like this:

[INFO] Executing: cmd.exe /X /C "git push https://user:secret@devserver/gitblit//r/waffel/devopts.git test-branch"

It should be avoided to ever print out passwords on the command line. I have encrypted the password in maven settings.xml ... but now it comes back and anybody can see them (also on a continues build server which should push with a dedicated user to a central repo).

---

1 votes, 4 watchers

**Comment by jira-importer** [comment]
**[Thomas Wabner](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=waffel)** commented

After some more investigation, I guess the problem ist in plexus-utils-3.0.15 in org.codehaus.plexus.util.cli.Commandline line 513 or (the cause) 534

The toString() method is triggered by org.apache.maven.scm.provider.git.gitexe.command.GitCommandLineUtils line 111.

There should be a change in this line to not use toString() ... here a "filtered" command line output should be used.

**Comment by jira-importer** [comment]
**[Ludovic Lebègue](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=llebegue)** commented

I have provided a patch for this one associated with a pull request on github :https://github.com/apache/maven-scm/pull/24

**Comment by jira-importer** [comment]
**[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy)** commented

Merged.
Thanks for the patch.

**Comment by jira-importer** [comment]
**[JQ](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jqassar)** commented

Hello,

The username and password are still being included in 1.9.4 when jgit is being used as the git provider:

```
build	10-Apr-2015 14:04:46	[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-scm-plugin:1.9.4:tag' with basic configurator -->
build	10-Apr-2015 14:04:46	[DEBUG]   (f) addTimestamp = false
build	10-Apr-2015 14:04:46	[DEBUG]   (f) basedir = /path/to/build/directory
build	10-Apr-2015 14:04:46	[DEBUG]   (f) connectionType = connection
build	10-Apr-2015 14:04:46	[DEBUG]   (s) connectionUrl = scm:git:https://host.address/scm/repo/repo.git
build	10-Apr-2015 14:04:46	[DEBUG]   (f) developerConnectionUrl = scm:git:https://host.address/scm/repo/repo.git
build	10-Apr-2015 14:04:46	[DEBUG]   (f) providerImplementations = {git=jgit}
build	10-Apr-2015 14:04:46	[DEBUG]   (f) pushChanges = true
build	10-Apr-2015 14:04:46	[DEBUG]   (f) remoteTagging = true
build	10-Apr-2015 14:04:46	[DEBUG]   (f) settings = org.apache.maven.execution.SettingsAdapter@213b0eef
build	10-Apr-2015 14:04:46	[DEBUG]   (f) tag = blah-0.0.99-24
build	10-Apr-2015 14:04:46	[DEBUG]   (f) timestampFormat = yyyyMMddHHmmss
build	10-Apr-2015 14:04:46	[DEBUG]   (f) timestampPosition = end
build	10-Apr-2015 14:04:46	[DEBUG]   (f) timestampPrefix = -
build	10-Apr-2015 14:04:46	[DEBUG] -- end configuration --
build	10-Apr-2015 14:04:46	[INFO] Change the default 'git' provider implementation to 'jgit'.
build	10-Apr-2015 14:04:46	[INFO] Final Tag Name: 'blah-0.0.99-24'
build	10-Apr-2015 14:04:47	[INFO] push tag [blah-0.0.99-24] to remote...
build	10-Apr-2015 14:04:47	[INFO] fetch url: https://foo:whoathere!@host.address/scm/repo/repo.git
build	10-Apr-2015 14:04:47	[INFO] push url: https://foo:whoathere!@host.address/scm/repo/repo.git
build	10-Apr-2015 14:04:48	[INFO] OK - RemoteRefUpdate[remoteName=refs/tags/blah-0.0.99-24, OK, (null)...a273fcdfa66fdcb925434fa6767038242d53d20d, fastForward, srcRef=refs/tags/blah-0.0.99-24, message=null]
```

**Comment by jira-importer** [comment]
**[Ludovic Lebègue](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=llebegue)** commented

An upgrade of maven-release-plugin to 2.5.2 solved the issue for me
